pulsar/producer_partition.go

```diff
@@ -236,6 +239,16 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 
 	msg := request.msg
 
+	// if msg is too large
+	if len(msg.Payload) > int(p.cnx.GetMaxMessageSize()) {
+		p.publishSemaphore.Release()
+		request.callback(nil, request.msg, errMessageTooLarge)
+		p.log.WithField("size", len(msg.Payload)).
+			WithField("properties", msg.Properties).
+			Error("message size exceeds MaxMessageSize")
```

Comment:
Oh, thanks for pointing this out~ 

As I'm new to this project, when I was writing these lines, I mainly referred to other error handling lines in the same function, which use `.Error("xxx")` rather than `.WithError()`. 

But I think your idea is better, so maybe we need to rewrite other error handling lines as well?

Reply:
I'm even newer to the project :) @wolfstudy ?

Reply:
Thanks @jinfengnarvar thanks for pointing this out.

Currently, there are some inconsistencies in the handling of errors. In order to ensure the structured output of the log, we should use `withError()` instead of `Error()`.